### PR TITLE
adding references to custom resources

### PR DESCRIPTION
**Pull request by dalbhanj**
Issue #, if available: https://github.com/aws-controllers-k8s/community/issues/1472

Description of changes: Once we add resource reference, emr-containers jobruns can reference virtualcluster custom resources directly. This is more k8s friendly and avoids exporting and importing variables

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: helm/templates/cluster-role-controller.yaml [hunk 1]
```diff
@@ -11,7 +11,6 @@ metadata:
   name: ack-emrcontainers-controller
   namespace: {{ .Release.Namespace }}
 {{ end }}
-rules:
```
Comment (A-Hilaly) [inline helm/templates/cluster-role-controller.yaml:14]: This is the cause of failing tests in https://github.com/aws-controllers-k8s/code-generator/pull/362 - investigating why this is happening. For now holding all the PRs using the latest version of code-generator

File: apis/v1alpha1/generator.yaml [hunk 2]
```diff
@@ -42,6 +42,10 @@ resources:
         is_immutable: true
       JobDriver:
         is_immutable: true
+      virtualClusterID:
```
Comment (RedbackThomson) [inline apis/v1alpha1/generator.yaml:45]: This actually needs to match the [AWS SDK name](https://docs.aws.amazon.com/sdk-for-go/api/service/emrcontainers/#StartJobRunInput) - so `VirtualClusterId`

File: apis/v1alpha1/generator.yaml [hunk 3]
```diff
@@ -42,6 +42,10 @@ resources:
...
+        references:
+          resource: VirtualCluster
+          path: Status.Id
```
Comment (RedbackThomson) [inline apis/v1alpha1/generator.yaml:48]: And confusingly, this needs to match the [CRD path](https://github.com/aws-controllers-k8s/emrcontainers-controller/blob/main/apis/v1alpha1/virtual_cluster.go#L58). So `Status.ID`

**Review by RedbackThomson — CHANGES_REQUESTED** [review]

File: apis/v1alpha1/generator.yaml (hunk 3 above)
Comment (dalbhanj) [reply apis/v1alpha1/generator.yaml]: fixed

File: apis/v1alpha1/generator.yaml (hunk 2 above)
Comment (dalbhanj) [reply apis/v1alpha1/generator.yaml]: fixed

**Comment by A-Hilaly** [comment]
Hi @dalbhanj, you will need to publish `ack-test-emrcontainers-f020dcf6:latest` in the emrcontainers AWS testing account. Ideally provide a script or a makefile so anyone can reproduce tests on there accounts https://github.com/aws-controllers-k8s/lambda-controller/tree/main/test/e2e/resources/lambda_function

**Review by A-Hilaly — APPROVED** [review]
++ Thanks Peter!

**Pull request merged**